src/Servers/Kestrel/Core/src/Internal/Http3/QPack/IntegerDecoder.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.QPack
+{
+    /// <summary>
+    /// The maximum we will decode is Int32.MaxValue, which is also the maximum request header field size.
+    /// </summary>
+    internal class IntegerDecoder
```

Comment:
Same as HPACK `IntegerDecoder`?

Reply:
Yes, the encoder and decoders should be the same. I'll make them helpers.

Reply:
Err I'll share them is probably the better way to say it 😄